packages/web/src/scss/components/PricingPlan/_PricingPlanHeader.scss

```diff
@@ -0,0 +1,77 @@
+// 1. Use a subgrid layout to allow for the header items to be aligned vertically across multiple pricing plans.
+//
+//    👉 Please note that the pricing plans must be wrapped in the Matrix component for this to work. Otherwise, the
+//    header items will be just stacked vertically.
+//
+// 2. Since the badge at the top of the header is optional, we use an additional element for the header content, so we
+//    don't have to add `padding-inline` to individual header items.
+//
+// 3. Again, the additional container re-applies the subgrid layout to align the header items vertically across multiple
+//    pricing plans.
+//
+// 4. We use named grid lines (established by the parent subcomponent) to place the header and its items in the correct
+//    header rows so they are aligned vertically even when some of the header items are missing from any pricing plan.
+
+@use '../../tools/typography';
+@use 'theme';
+
+.PricingPlanHeader__badge,
+.PricingPlanHeader__content {
+    padding-inline: theme.$padding-x;
+}
+
+.PricingPlanHeader {
+    display: grid;
```

Comment:
You could also group class names to apply `display: grid`, but probably it's better to prefer readability as you did.

Reply:
You are right, that's why I have it like this: preference of readability over micro-optimization.